agent/discovery/sapsystem.go

```diff
@@ -26,28 +29,39 @@ func (d SAPSystemsDiscovery) GetId() string {
 	return d.id
 }
 
-func (d SAPSystemsDiscovery) Discover() error {
+func (d SAPSystemsDiscovery) Discover() (string, error) {
```

Comment:
The changes in this `Discover` will conflict with changes coming from: https://github.com/trento-project/trento/pull/160
I think we should take the change from that other PR, as the sid concatenation looks better

Reply:
Rebased!